### PR TITLE
[MU4] Slurs only adjust for slurs on their staff, and endpoint adjustment fix

### PR DESCRIPTION
**Pull request by asattely**
There were a few bugs in the slur adjustment code that caused slurs to adjust for slurs not on their system. This has been fixed. There was also a bit of a bug in the slur endpoint avoidance code which has been fixed now as well.

There still seems to be a strange issue with accidentals not being taken into account in the collision detection code (which this PR or the previous slur PR never touched), and I can't help but think that this issue didn't exist when the last slur revamp was written. So the shapewise collision stuff for accidentals still needs to be looked at a bit closelier in the future. This PR won't deal with that however!

### CODE REVIEW

**Comment by asattely** [comment]
vtest failures are expected, I believe (@oktophonie can confirm or deny)

**Comment by oktophonie** [comment]
The endpoints of slurs are not avoiding ties any more, when crossing a system:

slur-tie-collision

**Review by oktophonie — APPROVED** [review]

File: src/engraving/libmscore/slur.cpp [hunk 1]
```diff
@@ -1426,15 +1426,11 @@ SpannerSegment* Slur::layoutSystem(System* system)
             Chord* c = toChord(cr);
             Tie* tie = nullptr;
             PointF endPoint;
-            if (c->notes()[0]->tieFor() && !c->notes()[0]->tieFor()->isInside() && c->notes()[0]->tieFor()->up() == _up) {
-                // there is a tie that starts on this chordrest
-                tie = c->notes()[0]->tieFor();
-                endPoint = tie->segmentAt(0)->ups(Grip::START).pos();
-            }
-            /*if (c->notes()[0]->tieBack() && !c->notes()[0]->tieBack()->isInside() && c->notes()[0]->tieBack()->up() == _up) {
+            if (c->notes()[0]->tieBack() && !c->notes()[0]->tieBack()->isInside() && c->notes()[0]->tieBack()->up() == _up) {
```
Comment (RomanPudashkin) [inline src/engraving/libmscore/slur.cpp:1429]: Please, add a temp variable for c->notes()[0]->tieBack() instead of constantly copy-pasting it

**Comment by asattely** [comment]
Good call! The changes are pushed.

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**